prime-router/src/main/kotlin/metadata/Mappers.kt

```diff
@@ -108,6 +108,102 @@ class MiddleInitialMapper : Mapper {
+        if (args.size != 5) error("Schema Error: Invalid number of arguments - 5 required")
+        return args
+    }
+
+    /**
+     * Determines the meaning of a mapper argument
+     *
+     * Determines if a mapperArg exists a List of <ElementAndValue>s.
+     * If it does, the found element's .value is returned; otherwise, the argument itself is
+     * returned as a String literal.  This allows the mapper property of a schema to have literal
+     * arguments.  The downside is that misspelled or non-existent schema elements are treated
+     * as string literals.  Use with care.
+     *
+     * @param values the list of ElementAndValue objects passed to the apply function
+     * @param mapperArg one of the arguments from a schema elements mapper property
+     * @return Either a passed-in Elements .value or the mapper argument as a string literal
+     */
+    private fun decodeArg(values: List<ElementAndValue>, mapperArg: String): String {
+        return values.find { it.element.name.equals(mapperArg, ignoreCase = true) }?.value ?: mapperArg
+    }
+
+    /**
+     * Compares two strings for equalities
+     *
+     * If BOTH strings are numeric ("12"), it compares them as numbers.
+     * Otherwise they are compared as string literals
+     * @param op a String which denotes a comparison
+     * @param val1 a String literal which is either numeric or not
+     * @param val2 a String literal which is either numeric or not
+     * @return the Boolean result of properly applying the operator implied in op on val1 and val2
+     */
+    fun comp(op: String, val1: String, val2: String): Boolean {
+        val dVal1 = val1.toDoubleOrNull()
+        val dVal2 = val2.toDoubleOrNull()
+        if ((dVal1 == null) && (dVal2 != null)) {
+            error("ifThenElse Type Mismatch Error: $val1 is not numeric")
+        }
+        if ((dVal1 != null) && (dVal2 == null)) {
+            error("ifThenElse Type Mismatch Error: $val2 is not numeric")
+        }
+        if ((dVal1 != null) && (dVal2 != null)) { // only if both val1 and val2 are Double strings
+            return when (op) {
+                "==" -> (dVal1 == dVal2) // all Double comparisons (covers Float, Int)
+                "!=" -> (dVal1 != dVal2)
+                ">=" -> (dVal1 >= dVal2)
+                "<=" -> (dVal1 <= dVal2)
+                "<" -> (dVal1 < dVal2)
+                ">" -> (dVal1 > dVal2)
+                else -> error("ifThenElse Mapper Argument Error: not a valid operator: $op")
+            }
+        }
+        return when (op) {
+            "==" -> (val1 == val2) // all string comparisons
+            "!=" -> (val1 != val2)
+            ">=" -> (val1 >= val2)
+            "<=" -> (val1 <= val2)
+            "<" -> (val1 < val2)
+            ">" -> (val1 > val2)
+            else -> error("ifThenElse Mapper Argument Error: not a valid operator: $op")
+        }
+    } // comp(op, v1, v2):Boolean
```

Comment:
Remove this `//` comment.  May be seen as deleted code.

Reply:
right.